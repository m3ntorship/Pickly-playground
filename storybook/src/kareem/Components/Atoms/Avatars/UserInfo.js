import React from "react";

function UserInfo({ Name, Hours }) {
  return (
    <div className="flex-col ml-4">
      <span className="h-5 w-48 font-sans text-sm text-black font-bold block">
        {Name}
      </span>
      <span className="h-5 w-28 font-sans text-sm text-dark-grey block">
        {Hours} hours ago
      </span>
    </div>
  );
}

export default UserInfo;
