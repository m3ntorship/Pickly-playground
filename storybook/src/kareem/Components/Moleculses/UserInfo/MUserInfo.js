import React from "react";
import Avatar from "../../Atoms/Avatars/Avatar";
import Info from "../../Atoms/Avatars/UserInfo";

function MUserInfo({ status = "normal", size = "medieum", Name, Hours }) {
  return (
    <div className="flex items-center justify-start h-10">
      <Avatar size={size} status={status} />
      <Info Name={Name} Hours={Hours} />
    </div>
  );
}

export default MUserInfo;
