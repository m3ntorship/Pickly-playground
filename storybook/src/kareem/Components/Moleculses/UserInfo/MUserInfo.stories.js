import React from "react";
import MUserInfo from "./MUserInfo";

export default {
  title: "kareem/Moleculses/UserInfo",
  component: MUserInfo,
  argTypes: {
    Name: {
      control: {
        type: "text",
      },
    },
    Hours: {
      control: {
        type: "text",
      },
    },
    status: {
      control: {
        type: "radio",
        options: ["normal", "anonymous", "notfilled"],
      },
    },
    size: {
      control: {
        type: "select",
        options: ["large", "medieum", "small"],
      },
    },
  },
};

const Template = (args) => <MUserInfo {...args} />;
export const Input = Template.bind({});
Input.args = {
  Name: "Ahmad Ayoub",
  Hours: "2",
  status: "normal",
  size: "medieum",
};
