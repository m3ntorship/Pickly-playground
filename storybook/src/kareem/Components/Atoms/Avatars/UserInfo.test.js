import React from "react";
import renderer from "react-test-renderer";
import UserInfo from "./UserInfo";

describe("Render the UserInfo component", () => {
  it("Kareem: Render UserInfo ", () => {
    const tree = renderer
      .create(<UserInfo Name={"Kareem Elhosseny"} Hours={"3"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
