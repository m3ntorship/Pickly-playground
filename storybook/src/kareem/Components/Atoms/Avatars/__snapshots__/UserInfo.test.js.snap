// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the UserInfo component Kareem: Render UserInfo  1`] = `
<div
  className="flex-col ml-4"
>
  <span
    className="h-5 w-48 font-sans text-sm text-black font-bold block"
  >
    Kareem Elhosseny
  </span>
  <span
    className="h-5 w-28 font-sans text-sm text-dark-grey block"
  >
    3
     hours ago
  </span>
</div>
`;
